ambry-router/src/test/java/com.github.ambry.router/CryptoJobHandlerTest.java

```diff
@@ -232,35 +239,39 @@ public void testPendingDecryptJobs() throws InterruptedException, GeneralSecurit
     SecretKeySpec perBlobKey = kms.getRandomKey();
     List<DecryptJob> decryptJobs = new CopyOnWriteArrayList<>();
     for (int i = 0; i < testDataCount; i++) {
-      Pair<BlobId, ByteBuffer> randomData = getRandomBlob(referenceClusterMap);
-      cryptoJobExecutorService.submitJob(
-          new EncryptJob(randomData.getFirst(), randomData.getSecond(), perBlobKey, cryptoService, kms,
+      TestBlobData testData = getRandomBlob(referenceClusterMap);
+      cryptoJobHandler.submitJob(
+          new EncryptJob(testData.blobId.getAccountId(), testData.blobId.getContainerId(), testData.blobContent,
+              testData.userMetadata, perBlobKey, cryptoService, kms,
               (EncryptJob.EncryptJobResult encryptJobResult, Exception exception) -> {
-                Assert.assertEquals("BlobId mismatch ", randomData.getFirst(), encryptJobResult.getBlobId());
-                Assert.assertNotNull("Encrypted content should not be null", encryptJobResult.getEncryptedContent());
+                Assert.assertNotNull("Encrypted content should not be null",
+                    encryptJobResult.getEncryptedBlobContent());
                 Assert.assertNotNull("Encrypted key should not be null", encryptJobResult.getEncryptedKey());
-                decryptJobs.add(new DecryptJob(randomData.getFirst(), encryptJobResult.getEncryptedKey(),
-                    encryptJobResult.getEncryptedContent(), cryptoService, kms,
-                    (DecryptJob.DecryptJobResult decryptJobResult, Exception e) -> {
-                      Assert.assertEquals("BlobId mismatch ", randomData.getFirst(), decryptJobResult.getBlobId());
-                      if (e == null) {
-                        Assert.assertNull(
-                            "Exception shouldn't have been thrown to decrypt contents for " + randomData.getFirst(),
-                            exception);
-                        Assert.assertNotNull("Decrypted contents should not be null",
-                            decryptJobResult.getDecryptedContent());
-                        Assert.assertArrayEquals("Decrypted bytes and plain bytes should match",
-                            randomData.getSecond().array(), decryptJobResult.getDecryptedContent().array());
-                      } else {
-                        Assert.assertNotNull(
-                            "Exception should have been thrown to decrypt contents for " + randomData.getFirst(), e);
-                        Assert.assertTrue("Exception cause should have been GeneralSecurityException",
-                            e instanceof GeneralSecurityException);
-                        Assert.assertNull("Decrypted contents should have been null",
-                            decryptJobResult.getDecryptedContent());
-                      }
-                      decryptCallBackCount.countDown();
-                    }));
+                decryptJobs.add(new DecryptJob(testData.blobId, encryptJobResult.getEncryptedKey(),
```

Comment:
Same comments as above (make sure that asserts in child threads can fail the test, static import for Assert)